File: DSA/linked_list/get_node_value.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

a = Node('A')
b = Node('B')
c = Node('C')
d = Node('D')

a.next = b
b.next = c
c.next = d

def get_node(node, index):
    count = 0
    while node:
        if count == index:
            return node.val
        node = node.next
        count += 1
    return -1 

def get_node_rec(node, index):
    if node == None:
        return -1
    if index == 0:
        return node.val
    return get_node_rec(node.next, index-1)

if __name__ == '__main__':
    print('Getting node')
    print(get_node(a, 2))
    print('Getting node recursively')
    print(get_node_rec(a, 2))

File: dsa/linked_list/getNodeValue.js
class Node {
	constructor(val) {
		this.val = val
		this.next = null
	}
}

const a = new Node('A')
const b = new Node('B')
const c = new Node('C')
const d = new Node('D')

a.next = b
b.next = c
c.next = d

const getNode = (node, index) => {
	let count = 0
	while (node != null) {
		if (count === index) return node.val
		node = node.next
		count += 1
	}
	return -1
}

const getNodeRec = (node, index) => {
	if (node === null) return -1
	if (index === 0) return node.val
	return getNodeRec(node.next, index-1)
}

console.log('Getting node')
console.log(getNode(a, 2))
console.log('Getting node recursively')
console.log(getNodeRec(a, 2))
